Tools.Package/Program.cs

```diff
@@ -0,0 +1,259 @@
+        /// <returns>0 if successful.</returns>
+        public static async Task<int> Main(string[] args)
+        {
+            var debugOption = new Option<bool>(
+                name: "--debug",
+                description: "Enable debug logging.")
+            {
+                IsHidden = true
+            };
+
+            var workspaceArgument = new Argument<string>(
+                name: "directory",
+                description: "Directory containing the items for the package");
+
+            var outputDirectory = new Option<string>(
+                name: "--output",
+                description: "Directory where the package will be stored.")
+            {
+                IsRequired = true,
+                ArgumentHelpName = "OUTPUT_DIRECTORY"
+            };
+            outputDirectory.AddAlias("-o");
+
+            var packageName = new Option<string>(
+                name: "--name",
+                description: "Name for the package.",
+                getDefaultValue: () => "Package")
+            {
+                IsRequired = false,
+                ArgumentHelpName = "OUTPUT_NAME"
+            };
+            packageName.AddAlias("-n");
+
+            var rootCommand = new RootCommand("This .NET tool allows you to create dmapp and dmprotocol packages.");
+            rootCommand.AddGlobalOption(debugOption);
+            rootCommand.AddGlobalOption(outputDirectory);
+            rootCommand.AddGlobalOption(packageName);
+
+            var dmprotocolSubCommand = new Command("dmprotocol", "Creates a dmprotocol package based on a protocol solution")
+            {
+                workspaceArgument
+            };
+            dmprotocolSubCommand.SetHandler(ProcessDmProtocolAsync, workspaceArgument, outputDirectory, packageName, debugOption);
+
+            var dmappType = new Option<string>(
+                name: "--type",
+                description: "Type of dmapp package.")
+            {
+                IsRequired = true,
+            }.FromAmong("automation", "visio", "dashboard", "protocolvisio");
+            dmappType.AddAlias("-t");
+
+            var buildNumber = new Option<uint>(
+                name: "--build-number",
+                description: "The build number of a workflow run.")
+            {
+                ArgumentHelpName = "BUILD_NUMBER"
+            };
+            buildNumber.AddAlias("-bn");
+
+            var version = new Option<string>(
+                name: "--version",
+                description: "The version number for the artifact. This takes precedent before 'build-number'. Supported formats: 'A.B.C', 'A.B.C.D', 'A.B.C-suffix' and 'A.B.C.D-suffix'.")
+            {
+                ArgumentHelpName = "VERSION",
+            };
+            version.AddAlias("-v");
+
+            var protocolName = new Option<string>(
+                name: "--protocolName",
+                description: "The protocol name. Only applicable for the 'protocolvisio' type.")
```

Comment:
an alternative approach could be to create a subcommand for a protocol package so that this option is only applicable for this type

Reply:
As discussed we'll leave it for now and we can evaluate later on if extra sub commands would be needed.